### PR TITLE
refactor: rm `CallRequest`

### PR DESCRIPTION
**Pull request by onbjerg**
Depends on https://github.com/alloy-rs/alloy/pull/178

### CODE REVIEW

File: Cargo.toml
Comment (DaniPopes) [inline Cargo.toml:221]: most normal formatter change

File: crates/primitives/src/proofs.rs
Comment (mattsse) [inline crates/primitives/src/proofs.rs:103]: are we doing this again 😭😭😭😭😭😭😭😭😭😭😭

File: crates/rpc/rpc-types-compat/src/transaction/mod.rs
Comment (mattsse) [inline crates/rpc/rpc-types-compat/src/transaction/mod.rs:196]: now that I see that I think the rpc object should also use u256 because it's more common

I assume, u256 is more widely used

File: crates/rpc/rpc-types-compat/src/transaction/mod.rs
Comment (mattsse) [inline crates/rpc/rpc-types-compat/src/transaction/mod.rs:208]: this should be exhaustive

File: crates/rpc/rpc/src/eth/api/transactions.rs
Comment (mattsse) [inline crates/rpc/rpc/src/eth/api/transactions.rs:593]: why is this extra code needed?

if we dedupe types, then this:

let transaction = match request.into_typed_request(

should still work

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/rpc/rpc-types-compat/src/transaction/mod.rs
Comment (onbjerg) [reply crates/rpc/rpc-types-compat/src/transaction/mod.rs:196]: agree this is why i stopped porting, going to change it to u256 before continuing this pr

File: crates/rpc/rpc/src/eth/api/transactions.rs
Comment (onbjerg) [reply crates/rpc/rpc/src/eth/api/transactions.rs:593]: we dont have `into_typed_request` in alloy, typed request doesn't exist there so this was to remove `TransactionRequest` in reth which clashed with the one in alloy. but if u notice i left a comment to remove this before the pr is undrafted

File: crates/primitives/src/proofs.rs
Comment (onbjerg) [reply crates/primitives/src/proofs.rs:103]: bro i legit don't know why it does this. will fmt on linux before merging

File: crates/rpc/rpc/src/eth/api/transactions.rs
Comment (mattsse) [reply crates/rpc/rpc/src/eth/api/transactions.rs:593]: I did not notice 🙈 my bad
cool then disregard that

**Comment by onbjerg** [comment]
This can either be blocked by https://github.com/alloy-rs/alloy/pull/183 or we can merge and remove the inlining after

File: crates/rpc/rpc-types/src/eth/transaction/typed.rs
Comment (mattsse) [inline crates/rpc/rpc-types/src/eth/transaction/typed.rs:35]: these are reasonable, should really be u256

**Review by mattsse — APPROVED** [review]
lgtm, pending blockers

**Pull request merged**
